pkg/registry/etcd/ns_server.go

```diff
@@ -76,36 +77,45 @@ func (n *etcdNSRegistryServer) Register(ctx context.Context, request *registry.N
 func (n *etcdNSRegistryServer) watch(query *registry.NetworkServiceQuery, s registry.NetworkServiceRegistry_FindServer) error {
 	logger := log.FromContext(n.chainContext).WithField("etcdNSRegistryServer", "watch")
 
-	for {
-		watcher, err := n.client.NetworkservicemeshV1().NetworkServices(n.ns).Watch(s.Context(), metav1.ListOptions{})
-		if err != nil {
-			return err
-		}
+	var watchErr error
+	for watchErr == nil {
+		watchErr = func() error {
+			timeoutSeconds := int64(time.Minute / time.Second)
+			watcher, err := n.client.NetworkservicemeshV1().NetworkServices(n.ns).Watch(s.Context(), metav1.ListOptions{
+				TimeoutSeconds: &timeoutSeconds,
+			})
+			if err != nil {
+				return err
+			}
+			defer watcher.Stop()
```

Comment:
@Bolodya1997 As I can see we need only this line for fix the issue. Could you explain why do we need to add an anonymous funciton?

Reply:
Fixed.